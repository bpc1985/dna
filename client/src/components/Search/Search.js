import React, {useState} from "react";
import { DebounceInput } from "react-debounce-input";

import "./Search.scss";

export default function Search({searchTerm, onSearch}) {
  return (
    <div>
      <DebounceInput
        className="searchInput"
        name="searchInput"
        placeholder="Search your package"
        minLength={2}
        debounceTimeout={300}
        value={searchTerm}
        onChange={e => onSearch(e.target.value)}
      />
    </div>
  );
}
