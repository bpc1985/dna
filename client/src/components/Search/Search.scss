.searchInput {
  padding: 5px 10px;
  margin: 8px 0;
  width: 100%;
  display: block;
  border: 1px solid #ff007c;
  box-sizing: border-box;
  min-width: 180px;
}