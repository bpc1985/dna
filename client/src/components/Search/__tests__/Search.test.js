import { render, fireEvent, wait } from '@testing-library/react';
import React from "react";
import Search from '../Search';

const setup = (onSearch) => {
  const utils = render(
    <Search onSearch={onSearch} />
  );
  const input = utils.container.querySelector('input[name="searchInput"]')
  return {
    input,
    ...utils,
  }
}

describe('Search', () => {
  it('should render component', async () => {
    const onSearch = jest.fn();
    const { container, input } = setup(onSearch);
    expect(container).toMatchSnapshot();

    fireEvent.change(input, {target: {value: 'abcdef'}});
    expect(input.value).toBe('abcdef');
    await wait(() => {
      expect(onSearch).toHaveBeenCalledTimes(1);
      expect(onSearch).toHaveBeenCalledWith('abcdef');
    });
  });
});
