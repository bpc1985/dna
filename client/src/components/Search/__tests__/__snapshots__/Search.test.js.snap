// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render component 1`] = `
<div>
  <div>
    <input
      class="searchInput"
      name="searchInput"
      placeholder="Search your package"
      type="text"
      value=""
    />
  </div>
</div>
`;
